src/core/client/stream/tabs/Profile/Settings/DeleteAccount/Pages/ConfirmPage.tsx

```diff
@@ -0,0 +1,241 @@
+import { FORM_ERROR, FormApi, FormState } from "final-form";
+import { Localized } from "fluent-react/compat";
+import React, { FunctionComponent, useCallback } from "react";
+import { Field, Form } from "react-final-form";
+
+import { InvalidRequestError } from "coral-framework/lib/errors";
+import { colorFromMeta, ValidationMessage } from "coral-framework/lib/form";
+import { useMutation } from "coral-framework/lib/relay";
+import {
+  composeValidators,
+  required,
+  validateDeleteConfirmation,
+} from "coral-framework/lib/validation";
+import {
+  Button,
+  CallOut,
+  Flex,
+  FormField,
+  HorizontalGutter,
+  InputLabel,
+  PasswordField,
+  TextField,
+  Typography,
+} from "coral-ui/components";
+
+import RequestAccountDeletionMutation from "./RequestAccountDeletionMutation";
+
+import PageStepBar from "./Common/PageStepBar";
+
+import sharedStyles from "./Common/Page.css";
+import styles from "./ConfirmPage.css";
+
+interface Props {
+  step: number;
+  onCancel: () => void;
+  onProceed: () => void;
+}
+
+interface FormProps {
+  confirmation: string;
+  password: string;
+}
+
+const ConfirmPage: FunctionComponent<Props> = ({
+  step,
+  onCancel,
+  onProceed,
+}) => {
+  const requestAccountDeletion = useMutation(RequestAccountDeletionMutation);
+
+  const onSubmit = useCallback(
+    async (input: FormProps, form: FormApi) => {
+      try {
+        await requestAccountDeletion({
+          password: input.password,
+        });
+      } catch (err) {
+        if (err instanceof InvalidRequestError) {
+          return err.invalidArgs;
+        }
+
+        return {
+          [FORM_ERROR]: err.message,
+        };
+      }
+
+      form.reset();
+
+      onProceed();
+      return;
+    },
+    [onProceed]
+  );
+
+  const onCancelClicked = useCallback(() => {
+    onCancel();
+  }, [onProceed]);
+
+  const preventSubmit = (
+    formState: Pick<
+      FormState,
+      | "pristine"
+      | "hasSubmitErrors"
+      | "hasValidationErrors"
+      | "dirtySinceLastSubmit"
+    >
+  ) => {
+    const {
+      pristine,
+      hasValidationErrors,
+      hasSubmitErrors,
+      dirtySinceLastSubmit,
+    } = formState;
+    return (
+      pristine ||
+      hasValidationErrors ||
+      (hasSubmitErrors && !dirtySinceLastSubmit)
+    );
```

Comment:
Object destructuring is unnecessary.

Reply:
Also, this does not need to exist in the render function has it has no shared scope.